From 3a2b9e1a30d68b966e4ebb27c63589c83530624f Mon Sep 17 00:00:00 2001
From: Inki Dae <inki.dae@samsung.com>
Date: Wed, 24 Jul 2013 13:40:12 +0900
Subject: [PATCH 0433/1302] drm/exynos: consider common clock framework to g2d
 driver.

This patch just changes clk_enable/disable to
clk_prepare_enable/clk_disable_unprepare.

Signed-off-by: Inki Dae <inki.dae@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/gpu/drm/exynos/exynos_drm_g2d.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/exynos/exynos_drm_g2d.c b/drivers/gpu/drm/exynos/exynos_drm_g2d.c
index 310fbe8..3925e44 100644
--- a/drivers/gpu/drm/exynos/exynos_drm_g2d.c
+++ b/drivers/gpu/drm/exynos/exynos_drm_g2d.c
@@ -812,7 +812,7 @@ static void g2d_dma_start(struct g2d_data *g2d,
 						struct g2d_cmdlist_node, list);
 
 	pm_runtime_get_sync(g2d->dev);
-	clk_enable(g2d->gate_clk);
+	clk_prepare_enable(g2d->gate_clk);
 
 	writel_relaxed(node->dma_addr, g2d->regs + G2D_DMA_SFR_BASE_ADDR);
 	writel_relaxed(G2D_DMA_START, g2d->regs + G2D_DMA_COMMAND);
@@ -865,7 +865,7 @@ static void g2d_runqueue_worker(struct work_struct *work)
 					    runqueue_work);
 
 	mutex_lock(&g2d->runqueue_mutex);
-	clk_disable(g2d->gate_clk);
+	clk_disable_unprepare(g2d->gate_clk);
 	pm_runtime_put_sync(g2d->dev);
 
 	complete(&g2d->runqueue_node->complete);
-- 
1.8.3.2

